Examples/NN/example_MLP_DLR_airfoil.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+#
+# Example of MLP.
+#
+# Last revision: 09/10/2024
+
+import os, numpy as np, torch, optuna
+import pyLOM
+
+
+def load_dataset(fname,inputs_scaler,outputs_scaler):
+    '''
+    Auxiliary function to load a dataset into a pyLOM
+    NN dataset
+    '''
+    d  = pyLOM.Dataset.load(fname)
+    td = pyLOM.NN.Dataset(
+        variables_out  = (d["CP"],), 
+        variables_in   = d.xyz,
+        # to have each Mach and AoA pair just once. 
+        # To have all possible combinations, use [d.get_variable('AoA'), d.get_variable("Mach")]
+        parameters     = [[*zip(d.get_variable('AoA'), d.get_variable('Mach'))]], 
+        inputs_scaler  = inputs_scaler,
+        outputs_scaler = outputs_scaler,
+    )
+    return d, td
+
+def print_dset_stats(name,td):
+    '''
+    Auxiliary function to print the statistics
+    of a NN dataset
+    '''
+    x, y = td[:]
+    pyLOM.pprint(0,f'name={name} ({len(td)}), x ({x.shape}) = [{x.min(dim=0)},{x.max(dim=0)}], y ({y.shape}) = [{y.min(dim=0)},{y.max(dim=0)}]')
+
+
+## Set device
+device = pyLOM.NN.select_device("cpu") # Force CPU for this example, if left in blank it will automatically select the device
+
+
+## Load datasets and set up the results output
+BASEDIR = './DATA'
+CASESTR = 'NRL7301'
+RESUDIR = 'MLP_DLR_airfoil'
+pyLOM.NN.create_results_folder(RESUDIR)
+
+scaler     = pyLOM.NN.MinMaxScaler()
+_,td_train = load_dataset(os.path.join(BASEDIR,f'{CASESTR}_TRAIN.h5'),scaler,scaler)
+_,td_test  = load_dataset(os.path.join(BASEDIR,f'{CASESTR}_TEST.h5'),scaler,scaler)
+_,td_val   = load_dataset(os.path.join(BASEDIR,f'{CASESTR}_VAL.h5'),scaler,scaler)
+
+print_dset_stats('train',td_train)
+print_dset_stats('test', td_test)
+print_dset_stats('val',  td_val)
+
+
+## define the optimizer
+optimizer = pyLOM.NN.OptunaOptimizer(
+    optimization_params={
+        "lr": (0.00001, 0.01),  # fixed parameter
+        "n_layers": (1, 4),  # optimizable parameter,
+        "batch_size": (128, 512),
+        "hidden_size": (200, 400),
+        "p_dropouts": (0.1, 0.5),
+        "epochs": 50,
+    },
+    n_trials=25,
+    direction="minimize",
+    pruner=optuna.pruners.MedianPruner(n_startup_trials=5, n_warmup_steps=5, interval_steps=1),
+    save_dir=None,
+)
+
+model = pyLOM.NN.MLP(
+    input_size=4,
+    output_size=1,
+    hidden_size=256,
+    n_layers=2,
+    p_dropouts=0.15,
+)
+
+training_params = {
+    "epochs": 150,
+    'lr_scheduler_step': 1,
+    "optimizer_class": torch.optim.Adam,
+    "loss_fn": torch.nn.MSELoss(),
+    "print_rate_epoch": 1,
+    "num_workers": 6,
```

Comment:
Is this to make optuna work in parallel?

Reply:
No, it's the number of workers that will use the torch dataloader (https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader)
At the time, optuna is only working with one process, but it can be easily changed adding one parameter on the NN.optimizer module

Reply:
Then I don't understand why one would need 6 subprocesses to load the data, unless we are dealing with a huge dataset. I'll make some experiments and if I can reduce it to 0 better for the sake of the examples.

Reply:
For the examples i agree that we could reduce that parameter to 0, but as you said, we observed a bottleneck on the dataloader when training with a huge dataset. Increasing the number of workers improved the speed significantly.